substrate/frame/identity/src/lib.rs

```diff
@@ -651,15 +662,30 @@ pub mod pallet {
 		pub fn clear_identity(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
 			let sender = ensure_signed(origin)?;
 
+			// Clear subs, release `Subs` deposit and remove super identities
 			let (subs_deposit, sub_ids) = SubsOf::<T>::take(&sender);
-			let id = IdentityOf::<T>::take(&sender).ok_or(Error::<T>::NoIdentity)?;
-			let deposit = id.total_deposit().saturating_add(subs_deposit);
 			for sub in sub_ids.iter() {
 				SuperOf::<T>::remove(sub);
 			}
+			let released_amount = T::Balances::release(
+				&HoldReason::SubIdentities.into(),
+				&sender,
+				subs_deposit,
+				Precision::Exact,
+			)?;
+			debug_assert!(subs_deposit.is_zero() || !released_amount.is_zero());
+
+			// Clear identity and release `Identity` deposit.
+			let id = IdentityOf::<T>::take(&sender).ok_or(Error::<T>::NoIdentity)?;
+			let deposit = id.total_deposit();
```

Comment:
Here total deposit includes the deposit for judgements but judgements fess are held in `HoldReason::Judgement` isn't it?

Maybe we should simply do release all `Identity`, `SubIdentities` and `Judgement`.

Reply:
Just replaced `.total_deposit()` in favour of `.base_deposit()`, and `.judgements_deposit()` with no impact on tests.

Definitely this proves we need an additional test to assert that `Identity` and `Judgements` are being treated separately.